test/e2e/jmx/jmx_test.go

```diff
@@ -0,0 +1,364 @@
+		require.NoError(t, err, "Error getting CloudWatch Agent ConfigMap")
+		require.NotNil(t, configMap, "CloudWatch Agent ConfigMap not found")
+
+		cwConfig, exists := configMap.Data["cwagentconfig.json"]
+		require.True(t, exists, "cwagentconfig.json not found in ConfigMap")
+		require.Contains(t, cwConfig, `jmx`, "JMX configuration not found in cwagentconfig.json")
+
+		service, err := clientset.CoreV1().Services("amazon-cloudwatch").Get(context.TODO(), "cloudwatch-agent", metav1.GetOptions{})
+		require.NoError(t, err, "Error getting CloudWatch Agent Service")
+		require.NotNil(t, service, "CloudWatch Agent Service not found")
+
+		serviceAccount, err := clientset.CoreV1().ServiceAccounts("amazon-cloudwatch").Get(context.TODO(), "cloudwatch-agent", metav1.GetOptions{})
+		require.NoError(t, err, "Error getting CloudWatch Agent Service Account")
+		require.NotNil(t, serviceAccount, "CloudWatch Agent Service Account not found")
+	})
+}
+
+func testJMXResources(t *testing.T, clientset *kubernetes.Clientset) {
+	t.Run("verify_jmx_resources", func(t *testing.T) {
+		deploymentName := strings.TrimSuffix(filepath.Base(env.SampleApp), ".yaml")
+		pods, err := clientset.CoreV1().Pods("test").List(context.TODO(), metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("app=%s", deploymentName),
+			FieldSelector: "status.phase=Running",
+		})
+		require.NoError(t, err, "Error getting pods for deployment")
+		require.NotEmpty(t, pods.Items, "No pods found for deployment")
+
+		var jmxTargetSystem string
+		switch filepath.Base(env.AgentConfig) {
+		case "jvm_tomcat.json", "containerinsights.json":
+			jmxTargetSystem = "jvm,tomcat"
+		case "kafka.json":
+			jmxTargetSystem = "kafka"
+		}
+
+		requiredEnvVars := map[string]string{
+			"OTEL_EXPORTER_OTLP_PROTOCOL":            "http/protobuf",
+			"OTEL_METRICS_EXPORTER":                  "none",
+			"OTEL_LOGS_EXPORTER":                     "none",
+			"OTEL_TRACES_EXPORTER":                   "none",
+			"OTEL_AWS_JMX_EXPORTER_METRICS_ENDPOINT": "http://cloudwatch-agent.amazon-cloudwatch:4314/v1/metrics",
+			"OTEL_JMX_TARGET_SYSTEM":                 jmxTargetSystem,
+			"JAVA_TOOL_OPTIONS":                      " -javaagent:/otel-auto-instrumentation-java/javaagent.jar",
+		}
+
+		for _, container := range pods.Items[0].Spec.Containers {
+			for _, envVar := range container.Env {
+				if expectedValue, exists := requiredEnvVars[envVar.Name]; exists {
+					require.Equal(t, expectedValue, envVar.Value, fmt.Sprintf("Unexpected value for environment variable %s in container %s", envVar.Name, container.Name))
+					delete(requiredEnvVars, envVar.Name)
+				}
+			}
+		}
+
+		require.Empty(t, requiredEnvVars, "Not all required environment variables were found in the pod")
+	})
+}
+
+//------------------------------------------------------------------------------
+// Metric Test Functions
+//------------------------------------------------------------------------------
+
+func testTomcatMetrics(t *testing.T) {
```

Comment:
Similar comment to `testJMXResources`

Reply:
I moved `ValidateMetrics()` into `testutils.go`.